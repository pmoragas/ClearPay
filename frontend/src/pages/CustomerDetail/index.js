import React, {useEffect} from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getCustomer } from 'store/customer/actions';

import styles from './styles.module.scss';

const CharacterDetail = () => {
    const dispatch = useDispatch();
    const params = useParams();
    const detailId = params.detailId ? parseInt(params.detailId, 10) : undefined;
    const { characterDetail } = useSelector((state) => state.character);

    useEffect(() => {
        dispatch(getCustomer(detailId));
    }, [dispatch, detailId]);

    return (
        <div className={styles.container}>
            {characterDetail &&
                <div>patata</div>
            }

        </div>
    );
}

export default CharacterDetail;
