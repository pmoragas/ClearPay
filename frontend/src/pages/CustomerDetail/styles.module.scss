@import '~styles/colors';

.container {
    display: flex;
    flex-direction: column;
    height: 100%;

    .customerInfo {
        height: 40%;
        display: flex;


        .personalInfo {
            width: 50%;
            padding: 30px;
        }

        .economicalInfo {
            width: 50%;
            padding: 30px;
        }

        .dataPoint {
            display: flex;
            flex-direction: column;
            padding: 8px;

            .label {
                border-bottom: 1px solid black;
            }
        }
    }

    .wallets {
        height: 60%;
        padding: 30px;

        .rows {
            height: 30px;

            &:hover {
                background-color: rgba(black,0.1);
            }
        }
        table {
            width: 100%;
        }

        .transferButton {
            width: 100px;
            font-weight: 600;
            font-size: 16px;
            color: $white;
            text-align: center;
            background-color: black;
            border: unset;
            border-radius: .75rem;

            &:hover {
                opacity: 50%;;
            }
        }
    }
}