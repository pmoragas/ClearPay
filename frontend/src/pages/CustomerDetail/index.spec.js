import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen } from '@testing-library/react';
import CustomerDetail from './index';
const { customer } = require('mock/customer');

let mockCustomerSelector;

jest.mock('react-redux', () => ({
	useSelector: () => mockCustomerSelector,
    useDispatch: () => {
		return jest.fn();
	},
    useParams: () => {
		return "1";
	},
}));

describe('Customer', () => {
	test('Render Customer Page', () => {
        const history = createMemoryHistory();
        mockCustomerSelector = {customerDetail:customer, hasDataChanged: false, errorMsg: undefined};

        render(
            <Router history={ history }>
                <CustomerDetail/>
            </Router>);

		expect(screen.getAllByText('Transfer').length).toEqual(3);
		expect(screen.getByText('Jan')).toBeInTheDocument();
		expect(screen.getByText('Stannard')).toBeInTheDocument();
		expect(screen.getByText('jstannard0@g.co')).toBeInTheDocument();
		expect(screen.getByText('1125â‚¬')).toBeInTheDocument();
	});

	test('snapshot is ok', () => {
        const history = createMemoryHistory();
        mockCustomerSelector = {customerDetail: customer, hasDataChanged: false, errorMsg: undefined};
		const { container } = render(
            <Router history={ history }>
                <CustomerDetail/>
            </Router>);

		expect(container).toMatchSnapshot();
	});
});
