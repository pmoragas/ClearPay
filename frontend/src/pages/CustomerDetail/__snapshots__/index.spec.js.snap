// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer snapshot is ok 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="customerInfo"
    >
      <div
        class="personalInfo"
      >
        <div
          class="dataPoint"
        >
          <div
            class="label"
          >
            First name
          </div>
          <div
            class="data"
          >
            Jan
          </div>
        </div>
        <div
          class="dataPoint"
        >
          <div
            class="label"
          >
            Last name
          </div>
          <div
            class="data"
          >
            Stannard
          </div>
        </div>
        <div
          class="dataPoint"
        >
          <div
            class="label"
          >
            Email
          </div>
          <div
            class="data"
          >
            jstannard0@g.co
          </div>
        </div>
      </div>
      <div
        class="economicalInfo"
      >
        <div
          class="dataPoint"
        >
          <div
            class="label"
          >
            Total amount
          </div>
          <div
            class="data"
          >
            1125€
          </div>
        </div>
      </div>
    </div>
    <div
      class="wallets"
    >
      <div>
        <label
          class="search-label"
          for="search-bar-0"
        >
          <span
            class="sr-only"
            id="search-bar-0-label"
          />
          <input
            aria-labelledby="search-bar-0-label"
            class="form-control "
            id="search-bar-0"
            placeholder="Search"
            type="text"
            value=""
          />
        </label>
        <hr />
        <div
          class="react-bootstrap-table"
        >
          <table
            class="table table-bordered"
          >
            <thead>
              <tr>
                <th
                  style="width: 80%; text-align: left; text-transform: none;"
                  tabindex="0"
                >
                  Wallet ID
                </th>
                <th
                  style="width: 10%; text-align: left; text-transform: none;"
                  tabindex="0"
                >
                  Amount
                </th>
                <th
                  style="width: 10%; text-align: center; text-transform: none;"
                  tabindex="0"
                >
                  
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="rows"
              >
                <td>
                  aasljdhalsdjhaskdjhahsd
                </td>
                <td>
                  375€
                </td>
                <td
                  style="text-align: center;"
                >
                  <button
                    class="transferButton"
                  >
                    Transfer
                  </button>
                </td>
              </tr>
              <tr
                class="rows"
              >
                <td>
                  akjsdkajshdkasd9a76d
                </td>
                <td>
                  750€
                </td>
                <td
                  style="text-align: center;"
                >
                  <button
                    class="transferButton"
                  >
                    Transfer
                  </button>
                </td>
              </tr>
              <tr
                class="rows"
              >
                <td>
                  asidjalsikda9s8d76ad
                </td>
                <td>
                  0€
                </td>
                <td
                  style="text-align: center;"
                >
                  <button
                    class="transferButton"
                  >
                    Transfer
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
